doc/source/users_guide/cime-customize.rst

```diff
@@ -0,0 +1,84 @@
+.. _customizing-cime:
+
+===========================
+CIME config and hooks
+===========================
+
+CIME provides the ability to define model specific config and hooks.
+
+The config alters CIME's runtime and the hooks are triggered during their event.
+
+-----------------------------------
+How does CIME load customizations?
+-----------------------------------
+
+CIME will search ``cime_config/customize`` and load any python found under this directory or it's children.
+
+Any variables, functions or classes loaded are available from the ``CIME.customize`` module.
+
+---------------------------
+CIME config
+---------------------------
+
+Available config and descriptions.
+
+=================================  ================  ====  =============================================================================================================================================================
+Variable                           Default           Type  Description                                                                                                                                                  
+=================================  ================  ====  =============================================================================================================================================================
+allow_unsupported                  True              bool  Allows creation of case that is not test or supported.
+archive_dart_component             True              bool  Archives dart component.
+archive_drv_component              True              bool  Archives drv component.
+baseline_store_teststatus          True              bool  Upon the completion of a SystemTest and GENERATE_BASELINE is True, the TestStatus will be copied from the case directory to the baseline directory.
+build_cime_component_lib           True              bool  Build CIME component lib.
+build_model_use_cmake              False             bool  When building model use CMake.
+calculate_mode_build_cost          False             bool  Calculates model build cost rather than using static value in test_scheduler
+case_setup_generate_namelist       False             bool  Creates namelist during case.setup for some tests.
```

Comment:
@jedwards4b For tests namelists are always generated in `case.setup`. This was an E3SM specific option to always generate namelists during `case.setup` (https://github.com/ESMCI/cime/blob/a00a6ed2fb2e8a8dceedc3973585d7cc3d867707/CIME/case/case_setup.py#L453-L456).

This could use a better name and description, I will update.

Reply:
@jasonb5 , I wish I could remember why this was added for e3sm. For what it's worth, I do think we want to keep this since it makes conceptual sense to call the component buildnml scripts, which essential are configure scripts, at case.setup time.